File: cli/clients/palimpsest_license.ts
// SPDX-License-Identifier: PMPL-1.0

import { postJson } from "../http.ts";
import type { HttpOptions } from "../types.ts";

/** Audit licenses for a package path with palimpsest-license service */
export async function audit(
  path: string,
  baseUrl: string,
  token: string | undefined,
  opts: HttpOptions,
): Promise<void> {
  const body = JSON.stringify({ path });
  await postJson(baseUrl, "/audit", token, body, opts);
}

File: crates/opm-cli/src/clients/palimpsest_license.rs
// SPDX-License-Identifier: PMPL-1.0

use anyhow::Result;

use crate::clients::http;

pub fn audit(path: &str, base_url: &str, token: Option<&str>, opts: &http::HttpOptions) -> Result<()> {
    let body = serde_json::json!({ "path": path }).to_string();
    http::post_json(base_url, "/audit", token, &body, opts)
}
